src/core/bloom.h

```diff
@@ -17,36 +18,83 @@ class Bloom {
   Bloom& operator=(const Bloom&) = delete;
 
  public:
+  Bloom() = default;
+
+  /// @brief Destroy must be called before calling the d'tor
+  ~Bloom();
+
   /**
-   * @brief Construct a new Bloom object
+   * @brief Initializes a new Bloom object
    *
    * @param entries - entries are silently rounded up to the minimum capacity.
-   * @param error must be in (0, 1) range.
+   * @param fp_prob - False-positive probability of collision. Must be in (0, 1) range.
    * @param heap
    */
-  Bloom(uint32_t entries, double error, mi_heap_t* heap);
-  ~Bloom();
+  void Init(uint64_t entries, double fp_prob, PMR_NS::memory_resource* resource);
+
+  /**
+   * @brief Destroys the object, must be called before destructing the object.
+   *
+   * @param resource - resource with which the object was initialized.
+   */
+  void Destroy(PMR_NS::memory_resource* resource);
 
-  Bloom(Bloom&&) = default;
-  Bloom& operator=(Bloom&&) = default;
+  Bloom(Bloom&& o);
 
   bool Exists(std::string_view str) const;
 
+  // Equivalent to the Exist above but accepts two fingerprints of the item.
+  bool Exists(uint64_t fp[2]) const;
+
   /**
    * @brief Adds an item to the bloom filter.
    * @param str  -
    * @return true if element was not present and was added,
    * @return false - if element (or a collision) had already been added previously.
    */
   bool Add(std::string_view str);
+  bool Add(uint64_t fp[2]);
```

Comment:
so now we need it only for tests 🤔 A TestHash: string -> fp[2] would solve all of it

Reply:
No, we do not use it for tests. We use it inside SBF?

Reply:
I mean the string version

Reply:
it's true that we won't not use Bloom directly in prod code but still it seems weird to omit this API